x/ibc/02-client/types/keys.go

```diff
@@ -18,67 +20,102 @@ const (
 // store the consensus state of a client.
 func ConsensusStatePath(clientID string) string {
-	return fmt.Sprintf("clients/%s/consensusState", clientID)
+	return string(KeyConsensusState(clientID))
 }
 
 // RootPath takes an Identifier and returns a Path under which to
 // store the root for a particular height of a client.
 func RootPath(clientID string, height uint64) string {
-	return fmt.Sprintf("clients/%s/roots/%d", clientID, height)
+	return string(KeyRoot(clientID, height))
 }
 
 // CommitterPath takes an Identifier and returns a Path under which
 // to store the committer of a client at a particular height
 func CommitterPath(clientID string, height uint64) string {
-	return fmt.Sprintf("clients/%s/committer/%d", clientID, height)
+	return string(KeyCommitter(clientID, height))
 }
 
 // KeyClientState returns the store key for a particular client state
 func KeyClientState(clientID string) []byte {
-	return []byte(ClientStatePath(clientID))
+	return append(
+		ibctypes.KeyPrefixBytes(ibctypes.KeyClientPrefix),
+		[]byte(clientStatePath(clientID))...,
+	)
 }
 
 // KeyClientType returns the store key for type of a particular client
 func KeyClientType(clientID string) []byte {
-	return []byte(ClientTypePath(clientID))
+	return append(
+		ibctypes.KeyPrefixBytes(ibctypes.KeyClientTypePrefix),
+		[]byte(clientTypePath(clientID))...,
+	)
 }
 
 // KeyConsensusState returns the store key for the consensus state of a particular
 // client
 func KeyConsensusState(clientID string) []byte {
-	return []byte(ConsensusStatePath(clientID))
+	return append(
+		ibctypes.KeyPrefixBytes(ibctypes.KeyConsensusStatePrefix),
+		[]byte(consensusStatePath(clientID))...,
+	)
 }
 
 // KeyRoot returns the store key for a commitment root of a particular
 // client at a given height
 func KeyRoot(clientID string, height uint64) []byte {
-	return []byte(RootPath(clientID, height))
+	return append(
+		ibctypes.KeyPrefixBytes(ibctypes.KeyRootPrefix),
+		[]byte(rootPath(clientID, height))...,
+	)
 }
 
-// KeyValidatorSet returns the store key for a validator of a particular
-// client at a given height. Key => clientID||height
+// KeyCommitter returns the store key for a validator (aka commiter) of a particular
+// client at a given height.
 func KeyCommitter(clientID string, height uint64) []byte {
-	return []byte(CommitterPath(clientID, height))
+	return append(
+		ibctypes.KeyPrefixBytes(ibctypes.KeyCommiterPrefix),
+		[]byte(committerPath(clientID, height))...,
+	)
+}
+
+// GetClientKeysPrefix return the ICS02 prefix bytes
+func GetClientKeysPrefix(prefix int) []byte {
+	return []byte(fmt.Sprintf("%d/clients", prefix))
+}
+
+func clientStatePath(clientID string) string {
+	return fmt.Sprintf("clients/%s/state", clientID)
+}
+
+func clientTypePath(clientID string) string {
+	return fmt.Sprintf("clients/%s/type", clientID)
+}
+
+func consensusStatePath(clientID string) string {
+	return fmt.Sprintf("clients/%s/consensusState", clientID)
```

Comment:
If we're using prefix bytes anyways the ends of these paths are redundant & we should remove.

I'll update the ICS specs in [ICS 7](https://github.com/cosmos/ics/pull/332).

Reply:
We can also use the ends at the beginning of the path as the prefixes. Up to you (I personally like that the full path is descriptive enough so that it reflects what the key’s value is)

Reply:
It is descriptive, but it's redundant & users don't read these paths anyways.

Reply:
> (I suppose by that logic the path shouldn't be human-readable at all though)

Agree

Reply:
Aye - instead let's do human-readable paths with non-conflicting prefixes.